### ISSUE TITLE
Sync ownCloud's tags with local desktop/OS

### ISSUE THREAD

**Issue by Bugsbane** [owncloud/client#4511]
With ownCloud 9+ now having a tag framework, this seems like a good time to look at syncing tags with the various operating systems / desktop environments local tag systems (such as that found in MacOS X, KDE 4+ and presumably Windows/gnome, so we're not losing this data when files are synced, and can also use / search for those tags across the platforms used.

This is related to https://github.com/owncloud/client/issues/1159 but goes beyond MacOS. Personally I'm more interested in plasma 5 integration, but to me it seems like regardless of OS, this is the same feature.

**Comment by elmcrest** [comment]
+1

**Comment by marbetschar** [comment]
+1

**Comment by Bugsbane** [comment]
Hey @elmcrest / @marbetschar ,
Thanks for your support of this issue. Just so you know, to add your support without spamming developers with notifications, most devs generally prefer that you use the "Add your reaction" smiley face button in the top right corner of the original post. This lets them see at a glance how many people added their support without needing to count comments, and it also keeps their inboxes free of a million "+1" emails that don't add actionable information.

Thanks again!

**Comment by marbetschar** [comment]
@Bugsbane thanks for the feedback; of course - your right ;)

Are there any news/plans regarding sync of tags to the OS filesystem? I stumble across this every few weeks, since nearly every time I use ownCloud I have to search through the files. Which I do on my local harddrive for productivity reasons. This is how a typical use case on my end looks:

I need this `Coyote` product logo. The product is from this `ACME` company. So lets look around all the following folders - since it wasn't me who created & stored the logo in the ownCloud:

- `ACME/Marketing/Coyote.png`
- `ACME/Coyote.png`
- `ACME/Coyote/logo.png`
- `ACME/Coyote/Marketing/logo.png`
- or anywhere else...?

Would be awesome to use the native Tags feature on macOS to simply grab all files tagged with `ACME`,`Logo`,`Coyote` !!

Sure, this is a simple example which can be organised by company wide rules.
But there are a LOT of cases, where you have to decide for one folder between a bunch of fully reasonable ones. Even worse, this decision is different for every person - hell the decision even differs for only one person depending on the time it is made.

Folders are just soooooo 1950.

**Comment by marbetschar** [comment]
any news on this enhancement?

**Comment by juangamnik** [comment]
+1. Anything new? Just tried to upload a file from macOS with tags, but tags are gone...

**Comment by Bugsbane** [comment]
Haven't heard anything so far, and it's been 1.3 years now...

**Comment by michaelstingl** [comment]
Some things are still missing on the server-side.

**Comment by falcongoat** [comment]
Could you elaborate what is needed to make this possible?

**Comment by Rene33** [comment]
I think it's not only tag, but all the extended attribute of the file. This is an important features of the files system on Mac OS, and some applications need these xattr to recognize their data.

**Comment by Bugsbane** [comment]
Well, could we *start* with tags and then build to the other extended file info later?

**Comment by Rene33** [comment]
Yes...or no. Tags are a convenient feature, you can use or not. xattr are used by some applications to recognize the files they use. And without them you cannot use the application. For instance my accounting application use the xattr. I am obliged to maintain a Dropbox sharing only for the files of the accounting application... 
Having said so, I don't know the complexity of both features. Aren't tags only a kind of xattr ?

**Comment by fablarosa** [comment]
Tags are stored in the com.apple.metadata:_kMDItemUserTags extended attribute.
You can check which tags are set using these commands when you are inside a directory containing tagged files:
`xattr -l *`
`ls -l@`
Tag syncing is a feature that should be definitely included in ownCloud.

**Comment by juangamnik** [comment]
Has anyone experience with using sparse bundle images in owncloud, in order to get this feature right now (you format the image with HFS+ which supports tags and since the sparse bundle splits the disk into 8MiB band files, the traffic penalty isn't that bad)?

**Comment by fablarosa** [comment]
Not so easy to use for the average user but it's a valid workaround, I did some quick testing and it looks like only the band files involved in file additions/modifications are syncronized by ownCloud.

**Comment by michaelstingl** [comment]
@fablarosa Thanks for checking. Please open a new issue if there is something that can be improved syncing _sparse bundle images_.

**Comment by fablarosa** [comment]
One major issue with sparse bundles I had not thought before is that they are image files, think them like iso images split in many small files.  
This would cause serious concurrency issues when mounting the same image file from two Macs and add/modify some files stored inside the sparse bundle, tested this myself right now and the results are unpredictable.

**Comment by guruz** [comment]
> One major issue with sparse bundles I had not thought before is that they are image files, think them like iso images split in many small files.

https://github.com/owncloud/client/issues/3202

**Comment by juangamnik** [comment]
> This would cause serious concurrency issues when mounting the same image file from two Macs and add/modify some files stored inside the sparse bundle, tested this myself right now and the results are unpredictable.

Reducing the band-size (let's say to 1 MB) should reduce the conflicts, but it is no solution. I try to unmount the image before leaving a machine (and wait for sync), but this is error-prone. Unfortunately, during unmounting HFS seems to do some last cleanup-writes (perhaps journaling...), so you have to unmount it, before you can mount it on another machine. Last but not least, there are some files (beside the bands) that can cause conflicts even easier like the single `info.plist`.

**Comment by elevatingcreativity** [comment]
It seems the thread got sidetracked a bit with the discussion of disk image files. For us, that's a non-starter. We have multiple team members sharing folders on owncloud, and image files would break due to the concurrency stuff, no matter how fine-grained. If one wrong byte gets flipped in an image, the whole image can be corrupted. Yes, I've had it happen...

As the OP pointed out, the way to reliably find things is definitely not relying on people to file them in the same ways. It is tagging.

However, I haven't seen any "official" action on this from the devs above....

Wouldn't this just be a matter of adding a column in the DB that stores the tags, and synchronize to that column from each client?

If I were more familiar with the way things work in this community, I might give it a go next month when I have more time - or at least make a contribution towards the effort. But I have no idea where I would start from getting such changes "approved" by the official releases even if I did so... 

Meanwhile, I'll hope this gets picked up by someone who could do this quickly.

**Comment by dpeger** [comment]
> Wouldn't this just be a matter of adding a column in the DB that stores the tags, and synchronize to that column from each client?

As ownCloud already supports tags in the backend and the web-frontend, I'd think that there is no need to perform any DB changes. From an outsiders position the looks more like a client issue (read OS/Fileysystem-Tags from files and submit them to the backend, receive tags from backend and store them in an OS/Filesystem dependent way). Not sure If the sync-protocol already supports meta-data though.

**Comment by aromicreativi** [comment]
_**file manager > right click > owncloud tags**_ (add and remove collaborative owncloud tags)

This would be a great feature expecially for image files, since you can use your file manager preview feature and quickly tag images. The browser interface is great to tag general files since you can recognize the content from the file name. However, image file names are meaningless since they are usually progressively set from a camera.

I submitted a feature request [here](https://github.com/owncloud/gallery/issues/730) in the gallery app issue tracker too, since that would be the natural image browsing interface.

Saving tags in extended attributes would be the proper way to ensure data portability.

**Comment by msrex** [comment]
Wir machen unser Meeting am Do wie geplant um rauszufinden was der Stand der Dinge ist. Dann sehen wir weiter.  Markus                                                                                                                                                                         -- Markus Rex - msrex@datto.com                                                                                                                                                                           From: tmatchett@datto.comSent: October 31, 2017 2:52 AMTo: msrex@datto.com; jstelter@datto.comSubject: Re: Work with peer group Synaxon  This reads like they want to distribute Datto; and I believe this to be a mistake. Distribution enhance / expands your business. However we’re in a situation where we’re not converting business. Synaxon will not help us with that! Distributors don’t sell; they take margin for increasing your volume of sales, but ultimately if we can’t close deals today this will not help. We need to first close deals before we look to increase volume (100% increase on zero, is still zero). We should make contact with them - but no drastic measures. Juergen - can you please forward the contact details of the person that gave you this?Thanks TrentOn Mon, 30 Oct 2017 at 22:36, Markus Rex <msrex@datto.com> wrote:

 
Jürgen,
 
I think we should by all means continue with them and see what needs to be done to be added to their portfolio. Do you have the contract to look at?
 
Best,
 
  Markus
 
On Friday, October 27, 2017 11:35:31 AM Jürgen Stelter wrote:

Hi Trent and Markus,

enclosed I have a presentation of Synaxon a peer group with more than 300 members.
From my point of view a valid group to work with in the future. 

They carry Autotask since July 2016. They sent us a contract earlier and we need to give them an update November 2nd how we go forward. 

Please review and let me know your thoughts and strategy. 

Kind regards, 

 

Jürgen Stelter
Channel Development Manager

Mob: +49 151 4070 2676
Datto DACHTHE SQUAIRE 1260547 Frankfurt, GermanyOffice: +49 69 6677 80901
www.datto.com/de/ 
Join the conversation!      

-- 
Markus Rex
<msrex@datto.com>-- Trent MatchettEMEA Sales DirectorDatto EMEA+ 44 (0) 7538 816 617www.datto.co.ukJoin the conversation!

**Comment by Bugsbane** [comment]
> file manager > right click > owncloud tags 

Personally I would prefer that we used the standard systems already in place for providing tags to files on each operating system rather than have our tags be a separate thing. Thing way users can tag with any app that used the operating standard, and have the sync client and OwnCloud/Nextcloud take care of keeping those intact across devices.

**Comment by cukabeka** [comment]
I agree with @Bugsbane - if the client could translate the OS tag to the owncloud tag, that would be awesome, but the approach from @aromicreativi would be a decent start IMHO. 
I still wonder why the enhancement label was removed by @SamuAlfageme ?

**Comment by guruz** [comment]
FYI this is what Dropbox syncs: https://help.dropbox.com/en-us/installs-integrations/sync-uploads/extended-attributes

**Comment by whiteonion** [comment]
I'd like to sponsor this feature!

**Comment by dwd0tcom** [comment]
... it's 2021 and this feature is still set as open?

**Comment by Bugsbane** [comment]
> ... it's 2021 and this feature is still set as open?

Yeah. And we still don't have flying cars and personal jetpacks, either. :(
